androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/FeatureDetector.java

```diff
@@ -0,0 +1,76 @@
+package com.google.androidbrowserhelper.trusted;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import androidx.browser.customtabs.CustomTabsService;
+
+/**
+ * Finds which features the given provider supports by querying PackageManager.
+ * To be replaced with an AndroidX API at some point.
+ */
+public class FeatureDetector {
+    private Map<String, SupportedFeatures> mSupportedFeaturesCache = new HashMap<>();
+
+    public boolean providerSupportsNavBarColorCustomization(Context context,
+            String providerPackage) {
+        return getSupportedFeatures(context, providerPackage).navbarColorCustomization;
+    }
+
+    public boolean providerSupportsColorSchemeParams(Context context, String providerPackage) {
+        return getSupportedFeatures(context, providerPackage).colorSchemeCustomization;
+    }
+
+    public boolean providerSupportsImmersiveMode(Context context, String providerPackage) {
+        return getSupportedFeatures(context, providerPackage).immersiveMode;
+    }
+
+    private SupportedFeatures getSupportedFeatures(Context context,
+            String providerPackage) {
+        SupportedFeatures cached = mSupportedFeaturesCache.get(providerPackage);
+        if (cached != null) return cached;
+
+        if (ChromeLegacyUtils.isChrome76(context.getPackageManager(), providerPackage)) {
```

Comment:
Could extend this to cover the other cases in ChromeLegacyUtils?

Reply:
Done